src/Dfe.PlanTech.AzureFunctions/Functions/QueueReceiver.cs

```diff
@@ -35,14 +35,50 @@ private async Task ProcessMessage(ServiceBusReceivedMessage message, ServiceBusM
         {
             try
             {
+                string cmsEvent = message.Subject.AsSpan()[(message.Subject.LastIndexOf('.') + 1)..].ToString();
                 var text = Encoding.UTF8.GetString(message.Body);
+
+                Logger.LogInformation("Performing Action: {action}", cmsEvent);
                 Logger.LogInformation("Processing {text}", text);
 
-                var mapped = _mappers.ToEntity(text);
+                ContentComponentDbEntity mapped = _mappers.ToEntity(text);
+                ContentComponentDbEntity? existing = GetExistingDbEntity(mapped);
+
+                if (existing != null)
+                {
+                    mapped.Archived = existing.Archived;
+                    mapped.Published = existing.Published;
+                    mapped.Deleted = existing.Deleted;
+                }
+
+                switch (cmsEvent)
```

Comment:
I'd bring line 38 (where you get the cmsEvent) and this switch statement out to its own function, but no biggie

Reply:
Resolved in: [fea93e7](https://github.com/DFE-Digital/plan-technology-for-your-school/pull/407/commits/fea93e7f76e5ed0c2efcc38162d9a5107ed3caa7)